src/dbnode/storage/index/block.go

```diff
@@ -533,7 +533,7 @@ func (b *block) addQueryResults(
 	batch []doc.Document,
 ) ([]doc.Document, int, int, error) {
 	// update recently queried docs to monitor memory.
-	if err := b.queryStats.Update(len(batch)); err != nil {
+	if err := b.queryStats.UpdateDocs(len(batch)); err != nil {
```

Comment:
Is this check necessary now that we perform one higher up in the call stack?

Reply:
Yeah we need this because this is what is actually incrementing the vals on a per-block level. The other new place we check if at the query-level just so we don't even bother iterating the individual blocks